app/assets/javascripts/application.js

```diff
@@ -19,14 +7,17 @@ jQuery(function($) {
   $('#payment-form').submit(function(event) {
     var $form = $(this);
     $form.find('button').prop('disabled', true);
-    Stripe.card.createToken($form, stripeResponseHandler);
-    return false;
+    if ($(".credit-card")[0].style.display != "none"){
```

Comment:
You only need to set the stripe publishable key once, so I'd consider moving the bits where you're finding the `#stripe-publishable-key` input element into the document.ready handler, outside of the submit handler. 


An alternative solution would be to do something like this below, rather than creating an input and relying on jQuery to get the input's value, and then calling the stripe function.

In a slim file somewhere

```
- content_for :javascript do
  script type="text/javascript" src="https://js.stripe.com/v2/"
  javascript:
    Stripe.setPublishableKey("#{Rails.configuration.stripe[:publishable_key]}");
```